### PR TITLE
hv: vtd: code cleanup

### PR DESCRIPTION
**Pull request by binbinwu1**
Some code cleanup in vtd.c

Tracked-On: #4831
Signed-off-by: Binbin Wu <binbin.wu@intel.com>
Acked-by: Eddie Dong <eddie.dong@intel.com>

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
13 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/include/arch/x86/vtd.h func:iommu_ecap_sc offset:215<br> reason:Static procedure is not explicitly called in code analysed. : iommu_ecap_sc<br><2> config:part_config filename:/hypervisor/arch/x86/vtd.c func:dmar_set_intr_remap_table offset:669<br> reason:Pointer param should be declared pointer to const. : dmar_unit<br><3> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:dmar_set_intr_remap_table offset:669<br> reason:Pointer param should be declared pointer to const. : dmar_unit<br><4> config:part_config filename:/hypervisor/arch/x86/vtd.c func:dmar_set_root_table offset:718<br> reason:Pointer param should be declared pointer to const. : dmar_unit<br><5> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:dmar_set_root_table offset:718<br> reason:Pointer param should be declared pointer to const. : dmar_unit<br><6> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:iommu_attach_device offset:1047<br> reason:Pointer not checked for null before use. : context_entry<br><7> config:part_config filename:/hypervisor/arch/x86/vtd.c func:iommu_attach_device offset:1047<br> reason:Pointer not checked for null before use. : context_entry<br><8> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:iommu_detach_device offset:1085<br> reason:Procedure contains UR data flow anomalies. : sid.bits.f<br><9> config:part_config filename:/hypervisor/arch/x86/vtd.c func:iommu_detach_device offset:1085<br> reason:Procedure contains UR data flow anomalies. : sid.bits.f<br><10> config:part_config filename:/hypervisor/arch/x86/vtd.c func:iommu_detach_device offset:1085<br> reason:Procedure contains UR data flow anomalies. : sid.bits.d<br><11> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:iommu_detach_device offset:1085<br> reason:Procedure contains UR data flow anomalies. : sid.bits.d<br><12> config:part_config filename:/hypervisor/arch/x86/vtd.c func:iommu_detach_device offset:1098<br> reason:Pointer not checked for null before use. : root_entry<br><13> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:iommu_detach_device offset:1098<br> reason:Pointer not checked for null before use. : root_entry<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/arch/x86/vtd.c [hunk 1]
```diff
@@ -732,26 +728,10 @@ static void dmar_invalid_iec_global(struct dmar_drhd_rt *dmar_unit)
 
 static void dmar_set_root_table(struct dmar_drhd_rt *dmar_unit)
```
Comment (dongyaozu) [inline hypervisor/arch/x86/vtd.c:718]: add @pre ?

**Comment by HuiHuangShi** [comment]
14 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:dmar_set_intr_remap_table offset:675<br> reason:Pointer param should be declared pointer to const. : dmar_unit<br><2> config:part_config filename:/hypervisor/arch/x86/vtd.c func:dmar_set_intr_remap_table offset:675<br> reason:Pointer param should be declared pointer to const. : dmar_unit<br><3> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:dmar_set_root_table offset:725<br> reason:Pointer param should be declared pointer to const. : dmar_unit<br><4> config:part_config filename:/hypervisor/arch/x86/vtd.c func:dmar_set_root_table offset:725<br> reason:Pointer param should be declared pointer to const. : dmar_unit<br><5> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:iommu_attach_device offset:1054<br> reason:Pointer not checked for null before use. : context_entry<br><6> config:part_config filename:/hypervisor/arch/x86/vtd.c func:iommu_attach_device offset:1054<br> reason:Pointer not checked for null before use. : context_entry<br><7> config:part_config filename:/hypervisor/arch/x86/vtd.c func:iommu_detach_device offset:1092<br> reason:Procedure contains UR data flow anomalies. : sid.bits.f<br><8> config:part_config filename:/hypervisor/arch/x86/vtd.c func:iommu_detach_device offset:1092<br> reason:Procedure contains UR data flow anomalies. : sid.bits.b<br><9> config:part_config filename:/hypervisor/arch/x86/vtd.c func:iommu_detach_device offset:1092<br> reason:Procedure contains UR data flow anomalies. : sid.bits.d<br><10> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:iommu_detach_device offset:1092<br> reason:Procedure contains UR data flow anomalies. : sid.bits.d<br><11> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:iommu_detach_device offset:1092<br> reason:Procedure contains UR data flow anomalies. : sid.bits.f<br><12> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:iommu_detach_device offset:1092<br> reason:Procedure contains UR data flow anomalies. : sid.bits.b<br><13> config:part_config filename:/hypervisor/arch/x86/vtd.c func:iommu_detach_device offset:1105<br> reason:Pointer not checked for null before use. : root_entry<br><14> config:sharing_config filename:/hypervisor/arch/x86/vtd.c func:iommu_detach_device offset:1105<br> reason:Pointer not checked for null before use. : root_entry<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

**Comment by shiqingg** [comment]
Add 'Coding Guidelines: PASS' label on behalf of Huihuang.

**Comment by binbinwu1** [comment]
@dongyaozu Could you help to review the PR?

**Pull request merged**